src/ngAnimate/animateJs.js

```diff
@@ -0,0 +1,243 @@
+      var animations = lookupAnimations(classes);
+      var before, after;
+      if (animations.length) {
+        var afterFn, beforeFn;
+        if (event == 'leave') {
+          beforeFn = 'leave';
+          afterFn = 'afterLeave';
+        } else {
+          beforeFn = 'before' + event.charAt(0).toUpperCase() + event.substr(1);
+          afterFn = event;
+        }
+
+        if (event !== 'enter' && event !== 'move') {
+          before = packageAnimations(element, event, options, animations, beforeFn);
+        }
+        after  = packageAnimations(element, event, options, animations, afterFn);
+      }
+
+      // no matching animations
+      if (!before && !after) return;
+
+      function applyOptions() {
+        options.$domOperation();
+        options.$applyClasses();
+      }
+
+      return {
+        start: function() {
+          var closeActiveAnimations;
+          var chain = [];
+
+          if (before) {
+            chain.push(function(fn) {
+              closeActiveAnimations = before(fn);
+            });
+          }
+
+          if (chain.length) {
+            chain.push(function(fn) {
+              applyOptions();
+              fn(true);
+            });
+          } else {
+            applyOptions();
+          }
+
+          if (after) {
+            chain.push(function(fn) {
+              closeActiveAnimations = after(fn);
+            });
+          }
+
+          var animationClosed = false;
+          var runner = new $$AnimateRunner({
+            end: function() {
+              endAnimations();
+            },
+            cancel: function() {
+              endAnimations(true);
+            }
+          });
+
+          $$AnimateRunner.chain(chain, onComplete);
+          return runner;
+
+          function onComplete(success) {
+            animationClosed = true;
+            applyOptions();
+            options.$applyStyles();
+            runner.complete(success);
+          }
+
+          function endAnimations(cancelled) {
+            if (!animationClosed) {
+              (closeActiveAnimations || noop)(cancelled);
+              onComplete(cancelled);
+            }
+          }
+        }
+      };
+    };
+
+    function executeAnimationFn(fn, element, event, options, onDone) {
+      var classesToAdd = options.addClass;
+      var classesToRemove = options.removeClass;
+
+      var args;
+      switch (event) {
+        case 'animate':
+          args = [element, options.from, options.to, onDone];
+          break;
+
+        case 'setClass':
+          args = [element, classesToAdd, classesToRemove, onDone];
+          break;
+
+        case 'addClass':
+          args = [element, classesToAdd, onDone];
+          break;
+
+        case 'removeClass':
+          args = [element, classesToRemove, onDone];
+          break;
+
+        default:
+          args = onDone ? [element, onDone] : [element];
```

Comment:
We still need to make the existing JS API work (which uses a done callback). Hopefully in 1.5 we can do something using `$animateRunner`.